src/main/java/ro/developmentfactory/thedrones/service/DroneServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package ro.developmentfactory.thedrones.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.developmentfactory.thedrones.entity.Drone;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+
+import java.sql.Timestamp;
+import java.sql.Date;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class DroneServiceImpl implements DroneServcie {
+
+    @Autowired
+    private DroneRepository droneRepository;
+    private DroneStatusService droneStatusService;
+
+    //Read operation
+    @Override public List<Drone> fetchDroneList(){
+        return (List<Drone>) droneRepository.findAll();
+    }
+
+    @Override
+    public Drone saveDrone(Drone drone) {
```

Comment:
maybe you should consider saving data to `drone_status` when you are creating the drone. I think we can assume that the drone is already on the field, so we just need to pick a default sector where all of them will appear during creation (maybe [0, 0]) and a default facing direction (depending on how you count your grid). if you do this, you could get the drone status immediately after it was created, which is probably logical

Reply:
done